File: backend/supabase_client.py
import os
from functools import lru_cache

from supabase import create_client, Client


@lru_cache
def get_supabase() -> Client | None:
    url = os.environ.get("SUPABASE_URL")
    key = os.environ.get("SUPABASE_SERVICE_ROLE_KEY")
    if not url or not key:
        return None
    return create_client(url, key)

File: lib/supabaseClient.ts
import { createClient } from "@supabase/supabase-js";

// The storefront currently reads from lib/products.ts (static seed data) so the
// site works out of the box with no Supabase project connected. Once your
// `products` table is populated (see supabase/schema.sql), swap the imports in
// app/shop, app/product/[slug] etc. from "@/lib/products" to fetch calls using
// this client — the field names already match the table columns.

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL;
const supabaseAnonKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY;

export const supabase =
  supabaseUrl && supabaseAnonKey ? createClient(supabaseUrl, supabaseAnonKey) : null;

export function isSupabaseConfigured() {
  return Boolean(supabase);
}
